container/container.go

```diff
@@ -723,12 +723,20 @@ func (container *Container) CreateDaemonEnvironment(tty bool, linkedEnv []string
 	if os == "" {
 		os = runtime.GOOS
 	}
-	env := []string{}
+
+	// Figure out what size slice we need so we can allocate this all at once.
+	envSize := len(container.Config.Env)
 	if runtime.GOOS != "windows" || (runtime.GOOS == "windows" && os == "linux") {
-		env = []string{
-			"PATH=" + system.DefaultPathEnv(os),
-			"HOSTNAME=" + container.Config.Hostname,
-		}
+		envSize += 2 + len(linkedEnv)
```

Comment:
not a blocker, but perhaps it would be good to have a comment where the `2` comes from; it's clear in this diff (`PATH` and `HOSTNAME`), but might cause some head-scratching in future

Reply:
Doesn't seem like too much head scratching anyway.
We should probably change how we do this anyway.